t/app/controller/admin_reportextrafields.t

```diff
@@ -0,0 +1,311 @@
+        $mech->get_ok("/report/new?longitude=-1.351488&latitude=51.847235&category=" . $contact->category);
+        $mech->content_lacks("this is a test description");
+        $mech->content_lacks("this field is a list");
+    };
+};
+
+FixMyStreet::override_config {
+    ALLOWED_COBRANDS => [ { 'noextras' => '.' } ],
+    MAPIT_TYPES => [ 'UTA' ],
+    MAPIT_URL => 'http://mapit.uk/',
+    LANGUAGES => [ 'en-gb,English,en_GB' ]
+}, sub {
+    subtest "Extra fields are missing from cobrand that doesn't allow them" => sub {
+        my $object = FixMyStreet::App->model('DB::ReportExtraFields')->first;
+        $object->update({ language => "", cobrand => ""});
+
+        $mech->get_ok("/report/new?longitude=-1.351488&latitude=51.847235&category=" . $contact->category);
+        $mech->content_lacks("this is a test description");
+        $mech->content_lacks("this field is a list");
+    };
+};
+
+FixMyStreet::App->model('DB::ReportExtraFields')->delete_all;
+$mech->log_out_ok;
+
+subtest 'Reports are created with correct extra metadata' => sub {
+    FixMyStreet::override_config {
+        ALLOWED_COBRANDS => [ 'tester' ],
+        MAPIT_URL => 'http://mapit.uk/',
+    }, sub {
+        my $model = FixMyStreet::App->model('DB::ReportExtraFields');
+        my $extra_fields = $model->find_or_create({
+            name => "Test extra fields",
+            language => "",
+            cobrand => ""
+        });
+        $extra_fields->push_extra_fields({
+            order => "1",
+            code => "string_test",
+            required => "true",
+            variable => "true",
+            description => "this is a test description",
+            datatype_description => "hint here",
+            datatype => "string",
+        });
+        $extra_fields->push_extra_fields({
+            order => "2",
+            code => "list_test",
+            required => "false",
+            variable => "true",
+            description => "this field is a list",
+            datatype_description => "",
+            datatype => "singlevaluelist",
+            values => [
+                { name => "name1", key => "key1" },
+            ]
+        });
+        $extra_fields->update;
+
+        my $user = $mech->create_user_ok('testuser@example.com', name => 'Test User');
+        $mech->log_in_ok($user->email);
+
+        $mech->get_ok('/report/new?latitude=55.952055&longitude=-3.189579');
+        $mech->content_contains($contact2->category);
+
+        my $extra_id = $extra_fields->id;
+        $mech->submit_form_ok( {
+            with_fields => {
+                title => "Test Report",
+                detail => "This is a test report",
+                category => $contact2->category,
+                "extra[$extra_id]string_test" => "Problem meta string",
+                "extra[$extra_id]list_test" => "key1",
+            }
+        } );
+        # my $content = $mech->content;
+        # $content =~ s/[^[:ascii:]]+//g;
+        # print $content;
```

Comment:
Few debug lines left in. (PS Try `utf8::encode($content)` for the middle line there :) )